### ISSUE TITLE
Consider moving back to custom GStreamer pipeline

### ISSUE THREAD

**Issue by fortaa** [mopidy/mopidy#1316]
Due to inconsistent behaviour of this class across platforms, which could result in suboptimal stream quality (e.g. Raspian and bit depth degradation),  relying  on more explicit gstream pipeline creation scheme are welcomed.

I guess there was an attempt already to switch to a non-playbin solution. Should the former one be an entry point for another try or completely new approach is favoured right now?

**Comment by adamcik** [comment]
Thanks for following up. For reference see also https://github.com/mopidy/mopidy/issues/1306#issuecomment-148544121

The old version of audio which doesn't use playbin is somewhere way back in the github history, could probably be dug up, but I'm unsure if it's worth building on. Gapless is a must going forward with something like this. I also want to make sure this happens only after the GStreamer 1.0 switch as it has a many playbin improvements already.

**Comment by yangshanjin** [comment]
Hi, does the mopidy depend on libspotify?

**Comment by kingosticks** [comment]
No. Mopidy-spotify depends on pyspotify which depends on libspotify.
Mopidy's dependencies are pretty clear in the documentation for installing
from source.

Hi, does the mopidy depend on libspotify?

—
Reply to this email directly or view it on GitHub
<https://github.com/mopidy/mopidy/issues/1316#issuecomment-150177895>.

**Comment by yangshanjin** [comment]
Firstly , thanks for your reply.I want to use libspotify in my embeded linux, but my cpu is mips architecture,  libspotify no support it, so I want to connect spotify with the help of mopity, is that  work？

**Comment by adamcik** [comment]
@yangshanjin please don't hijack this issue. For usage questions use https://discuss.mopidy.com/ where you can login with your github or other social account.

**Comment by kingosticks** [comment]
Please open a new question on discuss.mopidy.com rather than using this
thread as it is entirely unrelated.

**Comment by yangshanjin** [comment]
@kingosticks @adamcik Okay, I'm so sorry about that.

**Comment by adamcik** [comment]
https://gist.github.com/adamcik/cb3e96aa1adae003cd5c has a quick prototype with most of the kinks worked out, assumes GStreamer 1.2 or newer.

**Comment by jodal** [comment]
> *This was generated by AI during triage.*

The audio layer stays on playbin, and the current rebuild moves it to playbin3 and gives more explicit control through playbin's flags. Closing, because we do not plan to go back to a custom pipeline.
